frontend/src/pages/list/BookList.tsx

```diff
@@ -311,16 +261,28 @@ export default function BookList() {
     let publisher_only = false;
     let author_only = false;
     let isbn_only = false;
-    if (filterParams.filters.title.value) {
+    if (
+      "value" in filterParams.filters.title &&
+      filterParams.filters.title.value
+    ) {
       search_string = filterParams.filters.title.value;
       title_only = true;
-    } else if (filterParams.filters.publisher.value) {
+    } else if (
```

Comment:
can this be a switch case? or is it because of the && it can't.

Reply:
Yeah unfortunately the only way to make it type safe that I could figure out is like this. It's really long and a little messy, but before the change it was unfortunately too easy to accidentally change the name of a field in the Book interface, but not update the filter. This doesn't prevent that, but should help with clarity.